R/summarise_draws.R

```diff
@@ -124,23 +124,51 @@ summarise_draws.draws <- function(x, ..., .args = list()) {
   x <- repair_draws(x)
   x <- as_draws_array(x)
   variables <- variables(x)
-  out <- named_list(variables, values = list(named_list(names(funs))))
-  for (v in variables) {
-    draws <- drop_dims_or_classes(x[, , v], dims = 3, reset_class = TRUE)
-    args <- c(list(draws), .args)
-    for (m in names(funs)) {
-      out[[v]][[m]] <- do.call(funs[[m]], args)
-      if (rlang::is_named(out[[v]][[m]])) {
-        # use returned names to label columns
-        out[[v]][[m]] <- rbind(out[[v]][[m]])
-      }
+
+  # get length and output names, calculated on the first variable
+  v1 <- variables[1]
+  draws1 <- drop_dims_or_classes(x[, , v1], dims = 3)
```

Comment:
I suggest adding `reset_class = FALSE` as in the removed code.

Reply:
Yeah good catch.  This was just me rushing too fast while resolving conflicts.